tests/phpunit/integration/tests/Integrations/Site_Kit.php

```diff
@@ -54,6 +55,103 @@ public function test_register(): void {
 		$this->assertSame( 10, has_action( 'web_stories_print_analytics', [ $analytics, 'print_analytics_tag' ] ) );
 	}
 
+	/**
+	 * @covers ::register
+	 */
+	public function test_analytics_tracking_handler_webstories(): void {
+		// Init Analytics Module.
+		\define( 'GOOGLESITEKIT_VERSION', '1.2.3' );
+
+		// Update flexible switching option.
+		update_option( 'web_stories_ga_tracking_handler', 'web-stories', false );
+
+		// Get Analytics Module.
+		$analytics = $this->container->get( 'analytics' );
+		add_action( 'web_stories_print_analytics', array( $analytics, 'print_analytics_tag' ) );
+
+		// Register Site_Kit integration.
+		$this->instance->register();
+
+		// Check for filters & actions.
+		$this->assertSame(
+			10,
+			has_filter(
+				'googlesitekit_analytics-4_tag_amp_blocked',
+				function ( $blocked ) {
+					if ( $this->instance->context->is_web_story() ) {
+						return true;
+					}
+					return $blocked;
+				}
```

Comment:
@swissspidy Is this the correct way to verify static function? This doesn't seem right to me. Would you please suggest possible resolution on this?

Reply:
@swissspidy Added E2E unit tests and removed from PHPUnit.

Reply:
@swissspidy Moved from [E2E Unit Tests](https://github.com/GoogleForCreators/web-stories-wp/pull/13567/commits/82a7b4a4147494487e8b44948e70675cc99c35be) to [`<GoogleAnalyticsSettings />` component Unit Tests](https://github.com/GoogleForCreators/web-stories-wp/pull/13567/commits/383dee834e79fec6a7ed8ea3d7cf1a2ea3c1a95f).

Awaiting for your input on [updating `Note:` message](https://github.com/GoogleForCreators/web-stories-wp/blob/0634e91314aeffb5fa461d84240f0eb54f9efd3f/packages/wp-dashboard/src/components/editorSettings/googleAnalytics/index.js#L81-L84):
https://github.com/GoogleForCreators/web-stories-wp/blob/0634e91314aeffb5fa461d84240f0eb54f9efd3f/packages/wp-dashboard/src/components/editorSettings/googleAnalytics/index.js#L81-L84

cc: @AnuragVasanwala